lib/trento/infrastructure/operations/operations.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Trento.Infrastructure.Operations do
+  @moduledoc """
+  Operations integration
+  """
+
+  alias Trento.Operations.V1.{
+    OperationRequested,
+    OperationTarget
+  }
+
+  alias Trento.Infrastructure.Messaging
+
+  alias Trento.Infrastructure.Operations.AMQP.Publisher
+
+  require Logger
+
+  @type operation_target :: %{
+          agent_id: String.t(),
+          arguments: map()
+        }
+
+  @spec request_operation(String.t(), String.t(), String.t(), [operation_target()]) ::
+          :ok | {:error, any}
+  def request_operation(operation_id, group_id, operation, targets) do
+    operation_requested = %OperationRequested{
+      operation_id: operation_id,
+      group_id: group_id,
+      operation_type: operation,
+      targets:
+        Enum.map(targets, fn %{agent_id: agent_id, arguments: arguments} ->
+          %OperationTarget{agent_id: agent_id, arguments: map_value(arguments)}
+        end)
+    }
+
+    case Messaging.publish(Publisher, "requests", operation_requested) do
+      :ok ->
+        :ok
+
+      {:error, reason} = error ->
+        Logger.error("Failed to publish message to topic operations: #{inspect(reason)}")
+
+        error
+    end
+  end
+
+  defp map_value(map) when is_map(map) do
+    Enum.into(map, %{}, fn {key, value} -> {map_key(key), map_value(value)} end)
+  end
+
+  defp map_value(value) when is_number(value), do: %{kind: {:number_value, value}}
```

Comment:
Could we use for clarity the module `Trento.Operations.V1.OperationTarget.ArgumentsEntry` instead of the plain map?

Reply:
Of course to all the map value

Reply:
We cannot really use `ArgumentsEntry`. That's just an internal representation of a map given by the elixir protobuf implementation. In reality we still need to provide a plain map.
Either way, as commented offline, I found out that the new version of protobuf elixir library comes with a super util functions to abstract this logic: `to_map` and `from_map`